doc/source/serve/http-guide.md

```diff
@@ -1,124 +1,72 @@
-# HTTP with Serve
-
-
+# HTTP API
 
 This section should help you understand how to:
-
 - send HTTP requests to Serve deployments
 - use Ray Serve to integrate with FastAPI
 - use customized HTTP Adapters
-
-
-:::{note}
-HTTP Proxy HA is enabled by using [REST API](serve-in-production-deploying) or [Kubernetes operator](deploying-serve-on-kubernetes) to start the Ray Serve
-:::
+- choose which feature to use for your use case
 
 (serve-http)=
-
 ## Calling Deployments via HTTP
+When you deploy a Serve application, the ingress deployment (the one passed to `serve.run`) will be exposed over HTTP.
 
-When you deploy a Serve application, the ingress deployment (the one passed to `serve.run`) will be exposed over HTTP. If you want to route to another deployment, you can do so using the [ServeHandle API](serve-model-composition).
-
-```python
-@serve.deployment
-class Counter:
-    def __call__(self, request):
-        pass
+```{literalinclude} ../serve/doc_code/http_guide.py
+:start-after: __begin_starlette__
+:end-before: __end_starlette__
+:language: python
 ```
 
-Any request to the Serve HTTP server at `/` is routed to the deployment's `__call__` method with a [Starlette Request object](https://www.starlette.io/requests/) as the sole argument. The `__call__` method can return any JSON-serializable object or a [Starlette Response object](https://www.starlette.io/responses/) (e.g., to return a custom status code).
+Any request to the Serve HTTP server at `/` is routed to the deployment's `__call__` method with a [Starlette Request object](https://www.starlette.io/requests/) as the sole argument. The `__call__` method can return any JSON-serializable object or a [Starlette Response object](https://www.starlette.io/responses/) (e.g., to return a custom status code or custom headers).
 
-Below, we discuss some advanced features for customizing Ray Serve's HTTP functionality.
+Commonly, you just need the API to accept numpy array. You can use Serve's `DAGDriver` to simply the request parsing.
 
-(serve-fastapi-http)=
+```{literalinclude} ../serve/doc_code/http_guide.py
+:start-after: __begin_dagdriver__
+:end-before: __end_dagdriver__
+:language: python
+```
 
+```{note}
+Serve provides a library of HTTP adapters help you avoid boilerplate code. The [later section](serve-http-adapters) dive deeper into how it works.
```

Comment:
```suggestion
Serve provides a library of HTTP adapters help you avoid boilerplate code. The [later section](serve-http-adapters) dives deeper into how these work.
```

Reply:
Serve provides a library of HTTP adapters **to** help you